Sources/ScipioKit/Producer/Compiler.swift

```diff
@@ -0,0 +1,109 @@
+        self.extractor = DwarfExtractor(executor: executor)
+    }
+
+    private func buildArtifactsDirectoryPath(buildConfiguration: BuildConfiguration, sdk: SDK) -> URL {
+        rootPackage.workspaceDirectory.appendingPathComponent("\(buildConfiguration.settingsValue)-\(sdk.name)")
+    }
+
+    private func buildDebugSymbolPath(buildConfiguration: BuildConfiguration, sdk: SDK, target: ResolvedTarget) -> URL {
+        buildArtifactsDirectoryPath(buildConfiguration: buildConfiguration, sdk: sdk).appendingPathComponent("\(target).framework.dSYM")
+    }
+
+    func clean() async throws {
+        logger.info("🗑️ Cleaning \(rootPackage.name)...")
+        try await xcodebuild.clean(package: rootPackage)
+    }
+
+    func createXCFramework(target: ResolvedTarget,
+                           outputDirectory: URL) async throws {
+        let buildConfiguration = buildOptions.buildConfiguration
+        let sdks = extractSDKs(isSimulatorSupported: buildOptions.isSimulatorSupported)
+
+        let sdkNames = sdks.map(\.displayName).joined(separator: ", ")
+        logger.info("📦 Building \(target.name) for \(sdkNames)")
+
+        for sdk in sdks {
+            try await xcodebuild.archive(package: rootPackage, target: target, buildConfiguration: buildConfiguration, sdk: sdk)
+        }
+
+        logger.info("🚀 Combining into XCFramework...")
+
+        let debugSymbolPaths: [URL]?
+        if buildOptions.isDebugSymbolsEmbedded {
+            debugSymbolPaths = try await extractDebugSymbolPaths(target: target,
+                                                                 buildConfiguration: buildConfiguration,
+                                                                 sdks: sdks)
+        } else {
+            debugSymbolPaths = nil
+        }
+
+        try await xcodebuild.createXCFramework(
+            package: rootPackage,
+            target: target,
+            buildConfiguration: buildConfiguration,
+            sdks: sdks,
+            debugSymbolPaths: debugSymbolPaths,
+            outputDir: outputDirectory
+        )
+    }
+
+    private func extractDebugSymbolPaths(
+        target: ResolvedTarget,
+        buildConfiguration: BuildConfiguration,
+        sdks: Set<SDK>
+    ) async throws -> [URL] {
+        let debugSymbols: [DebugSymbol] = sdks.compactMap { sdk in
+            let dsymPath = buildDebugSymbolPath(buildConfiguration: buildConfiguration, sdk: sdk, target: target)
+            guard fileSystem.exists(dsymPath) else { return nil }
+            return DebugSymbol(dSYMPath: dsymPath,
+                               target: target,
+                               sdk: sdk,
+                               buildConfiguration: buildConfiguration)
+        }
+        // You can use AsyncStream
+        var symbolMapPaths: [URL] = []
+        for dSYMs in debugSymbols {
+            let maps = try await self.extractor.dump(dwarfPath: dSYMs.dwarfPath)
+            let paths = maps.values.map { uuid in
```

Comment:
I prefer to use more specific name than map

```suggestion
            let dumpedDSYMsMaps = try await self.extractor.dump(dwarfPath: dSYMs.dwarfPath)
            let paths = dumpedDSYMsMaps.values.map { uuid in
```